detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/ForbiddenAnnotationSpec.kt

```diff
@@ -0,0 +1,251 @@
+    }
+
+    @Test
+    fun `should report annotation on method`() {
+        val code = """
+        class SomeClass {
+            @SuppressWarnings("unused")
+            fun someMethod(){}
+        }
+        """.trimIndent()
+        val findings = ForbiddenAnnotation(
+            TestConfig(mapOf(ANNOTATIONS to listOf("java.lang.SuppressWarnings")))
+        ).compileAndLintWithContext(env, code)
+        assertThat(findings).hasSize(1).hasStartSourceLocation(2, 5)
+    }
+
+    @Test
+    fun `should report annotation on field`() {
+        val code = """
+        class SomeClass {
+            @SuppressWarnings("unused")
+            val someField: String = "lalala"
+        }
+        """.trimIndent()
+        val findings = ForbiddenAnnotation(
+            TestConfig(mapOf(ANNOTATIONS to listOf("java.lang.SuppressWarnings")))
+        ).compileAndLintWithContext(env, code)
+        assertThat(findings).hasSize(1).hasStartSourceLocation(2, 5)
+    }
+
+    @Test
+    fun `should report annotation on function parameter`() {
+        val code = """
+        fun main(@SuppressWarnings("unused") args: Array<String>){}
+        """.trimIndent()
+        val findings = ForbiddenAnnotation(
+            TestConfig(mapOf(ANNOTATIONS to listOf("java.lang.SuppressWarnings")))
+        ).compileAndLintWithContext(env, code)
+        assertThat(findings).hasSize(1).hasStartSourceLocation(1, 10)
+    }
+
+    @Test
+    fun `should report annotation on constructor`() {
+        val code = """
+        class SomeClass {
+            @SuppressWarnings("unused")
+            constructor(s: String)
+            constructor(t: Int)
+        }
+        """.trimIndent()
+        val findings = ForbiddenAnnotation(
+            TestConfig(mapOf(ANNOTATIONS to listOf("java.lang.SuppressWarnings")))
+        ).compileAndLintWithContext(env, code)
+        assertThat(findings).hasSize(1).hasStartSourceLocation(2, 5)
+    }
+
+    @Test
+    fun `should report annotation on local variable`() {
+        val code = """
+        fun some(): Int {
+            @SuppressWarnings("unused")
+            val q = "1234"
+            return 10
+        }
+        """.trimIndent()
+        val findings = ForbiddenAnnotation(
+            TestConfig(mapOf(ANNOTATIONS to listOf("java.lang.SuppressWarnings")))
+        ).compileAndLintWithContext(env, code)
+        assertThat(findings).hasSize(1).hasStartSourceLocation(2, 5)
+    }
+
+    @Test
+    fun `should report nested annotations`() {
+        val code = """
+        @Deprecated("unused", ReplaceWith("bar"))
+        fun foo() = "1234"
+        """.trimIndent()
+        val findings = ForbiddenAnnotation(
+            TestConfig(mapOf(ANNOTATIONS to listOf("kotlin.ReplaceWith")))
+        ).compileAndLintWithContext(env, code)
+        assertThat(findings).hasSize(1)
+            .hasStartSourceLocation(1, 23)
+            .hasTextLocations("ReplaceWith")
```

Comment:
I think this is a redundant suppression, since there's only one occurrence of ReplaceWith

```suggestion
            .hasTextLocations("ReplaceWith")
```

Reply:
Same for most tests, hasTextLocations might be superior to other numeric validations (hasStartSourceLocation/hasSize) if the text is unique. Curious what others think.

Reply:
I thought so too, but it was really frustrating to get an exception that the text from the argument appears more than once in a snippet. I thought it kinda confusing to have half of the tests with text and half with numbers.

Reply:
> half of the tests with text and half with numbers.

IMO that just highlights even more the type of the test.

Reply:
Nice one! It actually answers one thing I wasn't sure of, in form of a test. The highlight includes the @ symbol :)

Note: because of the @ symbol, it should be possible to use text matchers everywhere, no?

Reply:
Yes, indeed. Only one place remains where there are actually two of them (annotations for expressions), but I guess it's ok to leave only one expression for test. 

Thank you, sorry for me being so careless:(

Reply:
No worries, I'm learning with you :)